src/crewai/agent.py

```
@@ -264,14 +269,26 @@ def execute_task(
                     task=task,
                 ),
             )
-            result = self.agent_executor.invoke(
-                {
-                    "input": task_prompt,
-                    "tool_names": self.agent_executor.tools_names,
-                    "tools": self.agent_executor.tools_description,
-                    "ask_for_human_input": task.human_input,
-                }
-            )["output"]
+
+            # Determine execution method based on timeout setting
+            if self.max_execution_time is not None:
+                if not isinstance(self.max_execution_time, int) or self.max_execution_time <= 0:
+                    raise ValueError("Max Execution time must be a positive integer greater than zero")
+                result = self._execute_with_timeout(task_prompt, task, self.max_execution_time)
+            else:
+                result = self._execute_without_timeout(task_prompt, task)
+                
+        except TimeoutError as e:
+            # Propagate TimeoutError without retry
+            crewai_event_bus.emit(
+                self,
+                event=AgentExecutionErrorEvent(
+                    agent=self,
+                    task=task,
+                    error=f"Task '{task.description}' execution timed out after {self.max_execution_time} seconds. Consider increasing max_execution_time or optimizing the task.",
```

what about use the same `error` variable in the event error and exception message?

---

Actually I just notice that you can just propagate the error message 
```python
error = str(e)
```

---

Got it, changed,